packages/ui/src/domainModules/collectionMammals/components/MammalManager/index.js

```diff
@@ -131,31 +131,43 @@ const mapStateToProps = (
   )
 
   const mainColumnActiveTab = getMainColumnActiveTab(url)
-  const currentTableRowNumber = keyObjectGlobalSelectors.get.currentTableRowNumber(
-    state
-  )
+
   const totalNumberOfRecords =
     specimenSearchState &&
     specimenSearchState.items &&
     specimenSearchState.items.length
 
+  const currentTableRowNumber = keyObjectGlobalSelectors.get.currentTableRowNumber(
+    state
+  )
+
   const isEditRecordView = mainColumnActiveTab === 'recordEdit'
   const isNewRecordView = mainColumnActiveTab === 'recordNew'
   const isItemViewOrSettings = mainColumnActiveTab.startsWith('record')
   const isTableView = mainColumnActiveTab === 'resultTable'
   const isTableViewOrSettings = mainColumnActiveTab.startsWith('resultTable')
 
   const showSelectNextRecordButton =
-    !isNewRecordView && currentTableRowNumber !== totalNumberOfRecords
+    !isNewRecordView && currentTableRowNumber < totalNumberOfRecords
   const showSelectPreviousRecordButton =
-    !isNewRecordView && currentTableRowNumber !== 1
+    !isNewRecordView && currentTableRowNumber > 1
+
+  const focusedSpecimenId =
+    totalNumberOfRecords === 0
+      ? undefined
```

Comment:
this should not be necessary, handling this is the point of the change in line 418. I realize now that it is not possible to set the value to `undefined`, instead we have to delete the value.

you can trace this if you look at how the `set` action creators are created for the key object module: packages/ui/src/utilities/keyObjectModuleFactory/createActionCreators.js

and in the end you come to the setter in here, which will set an empty object if the value is undefined: packages/ui/src/utilities/stateHelper/immutableUtilities/index.js

see later comment with fix (I think)